### ISSUE TITLE
Default Exact_integer type without GMP/LEDA

### ISSUE THREAD

**Issue by afabri** [CGAL/cgal#3808]
## Issue Details

In  [`CGAL/Exact_integer.h`](https://github.com/CGAL/cgal/blob/master/Number_types/include/CGAL/Exact_integer.h#L38)  we say that we cannot do without a GMP or Leda, while `Exact_rational` is defined as `CGAL::Quotient<CGAL::MP_Float>`.  This seems not coherent to me, that is I would expect that `MP_Float` could be the built-in exact integer, but maybe I just miss something here. 

## Environment

* CGAL version:  master

**Comment by lrineau** [comment]
I agree. Both [`MP_Float`] and [`Exact_integer`] are documented to be model of `EuclideanRing` and `RealEmbeddable`.

[`MP_Float`]: https://doc.cgal.org/latest/Number_types/classCGAL_1_1MP__Float.html
[`Exact_integer`]: https://doc.cgal.org/latest/Number_types/group__nt__cgal.html#gae96d924c796bfc4e781d28f81df43c52

`MP_Float` is also a model of `FromDoubleConstructible`, but that does not prevent it from being an implementation of `CGAL::Exact_integer`.

**Comment by mglisse** [comment]
> In [`CGAL/Exact_integer.h`](https://github.com/CGAL/cgal/blob/master/Number_types/include/CGAL/Exact_integer.h#L38) we say that we cannot do without a GMP or Leda, while `Exact_rational` is defined as `CGAL::Quotient<CGAL::MP_Float>`. This seems not coherent to me, that is I would expect that `MP_Float` could be the built-in exact integer, but maybe I just miss something here.

`Quotient<MP_Float>` does have most of the properties of a rational number, including the most obvious one that it represents all rational numbers and only rational numbers. Well, `numerator` does not return an integer so it isn't perfect. `MP_Float` is much further from being an integer (in particular for division and remainder, but also simply because it includes 0.5). If all you want is a ring type that can represent all integers (I didn't check where exactly we use it), it is misleading to call it Exact_integer IMO, so there should at least be a large warning sign in the doc saying that it is **not** necessarily an integer type.
